### PR TITLE
RB506 effect code and placeholder sprite for vfx

### PR DESCRIPTION
**Pull request by EltonZhang777**
Completes code for RB506

Adds placeholder sprite for its vfx

Added the `PenitentSwordAttackEvent` in EventHandler

### CODE REVIEW

File: Blasphemous.LostDreams/Items/RosaryBeads/RB506.cs [hunk 1]
```diff
@@ -1,12 +1,401 @@
...
+        spriteObject.transform.SetParent(_projectiles[currentProjectileIndex].gameObject.transform);
+        spriteObject.AddComponent<SpriteRenderer>();
+
+        var rb = _projectiles[currentProjectileIndex].gameObject.GetComponent<Rigidbody2D>();
+        var collider = _projectiles[currentProjectileIndex].gameObject.GetComponent<BoxCollider2D>();
+        var sr = _projectiles[currentProjectileIndex].gameObject.GetComponentInChildren<SpriteRenderer>();
+
+        collider.size = new Vector2(2.8f, 1f);
+
+        Main.LostDreams.FileHandler.LoadDataAsSprite($"effects/Holy_Water_Projectile_Attack.png", out Sprite sprite);
```
[the remark this answers is not in the document]
Comment (BrandenEK) [reply Blasphemous.LostDreams/Items/RosaryBeads/RB506.cs:173]: Loading it in the item constructor would be fine

[the remark this answers is not in the document]
Comment (EltonZhang777) [reply Blasphemous.LostDreams/Items/RosaryBeads/RB506.cs:173]: Oh that's an obvious oversight that should have been taken care of. Thanks for pointing that out! Definitely will notice similar oversights in future.

**Comment by EltonZhang777** [comment]
All suggestions taken care of. 
Using an array for the projectiles was a design oversight, but could be actually useful if the cooldown of the projectiles are short enough or the projectiles last long enough time, so that they might be multiple projectiles in-game at once. Judging from this perspective, using array is actually more convenient for future code reuse on similar effects.

**Review by BrandenEK — APPROVED** [review]
Yep, everything has been resolved perfectly.  Very nice!

**Comment by BrandenEK** [comment]
> All suggestions taken care of. Using an array for the projectiles was a design oversight, but could be actually useful if the cooldown of the projectiles are short enough or the projectiles last long enough time, so that they might be multiple projectiles in-game at once. Judging from this perspective, using array is actually more convenient for future code reuse on similar effects.

I wouldn't call it an oversight, I just think it is more complicated than it needs to be.  It is useful if there are multiple projectiles at once, so it is good to leave it that way.

**Pull request merged**